### ISSUE TITLE
New Proposal for the way to extend stack config

### ISSUE THREAD

**Issue by GlieseRay** [Kotaimen/awscfncli#12]
For now, awscfncli uses `anchor` of YAML to reuse configuration shared with multiple stacks. It works however the syntax looks a little tricky and not very straight forward. So Here is a proposal for a new way to share configuration with multiple stacks. For example:

Suppose we have a Config template:

```yaml
StackConfigTemplate:
  Region: us-east-1
  Tags:
    Project:  Bob
    CostUnit: 180
```
And we are going to extend this template:

```yaml
Stack1:
  Extends: StackConfigTemplate
  Region: us-west-1
  Tags:
    CostUnit: 190
```

Then the result stack config will be:

```yaml
Stack1:
  Region: us-west-1
  Tags:
    Project:  Bob
    CostUnit: 190
```

General rule for extension:
1. Scalar: Replace
2. Array: Append
3. Dictionary: Merge

For properties such as `Capabilities`, special actions will be taken to ensure the completeness of extension. (For `Capabilities`, maybe replace is a more appropriate approach.)

**Comment by Kotaimen** [comment]
Looks good to me.

**Comment by Kotaimen** [comment]
Implemented in e9a6ce5.
